gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -504,12 +505,15 @@ public void handleLaunchFlowEvent(DagActionStore.DagAction action) {
       if (optionalJobExecutionPlanDag.isPresent()) {
         addDag(optionalJobExecutionPlanDag.get(), true, true);
       }
+      // Upon handling the action, delete it so on leadership change this is not duplicated
+      this.dagActionStore.get().deleteDagAction(action);
```

Comment:
So what happens here if the leader adds the dag and then dies before deletion? The subsequent leader startup would be adding the same dag again, is that intended behavior?

Reply:
Yes, for now the subsequent leader will add the same dag again. We have at least once guarantee with executions so this is pretty tolerable for now imo. The issue arises when service is overloaded and keeps dying for some reason before deletion. In average case, we won't really have too many dags with multiple execution if we also have retention running on this table.

Reply:
We can plan a refactor to save the compilation of the dag in mysql so we can de-dup dags with the same plan id and avoid this problem down the road.